src/tools/rbd_mirror/image_replayer/journal/Replayer.cc

```diff
@@ -1270,8 +1272,16 @@ void Replayer<I>::register_perf_counters() {
 
   auto cct = static_cast<CephContext *>(m_state_builder->local_image_ctx->cct);
   auto prio = cct->_conf.get_val<int64_t>("rbd_mirror_image_perf_stats_prio");
-  PerfCountersBuilder plb(g_ceph_context, "rbd_mirror_image_" + m_image_spec,
-                          l_rbd_mirror_journal_first, l_rbd_mirror_journal_last);
+
+  auto *local_image_ctx = m_state_builder->local_image_ctx;
+  std::string labels = ceph::perf_counters::key_create(
+      "rbd_mirror_image",
+      {{"pool", local_image_ctx->md_ctx.get_pool_name()},
+       {"image", local_image_ctx->name},
+       {"namespace", local_image_ctx->md_ctx.get_namespace()}});
```

Comment:
Swap this line with above line to make `namespace` go before `image` to maintain natural (larger-to-smaller) order.